update flutter analyze to call dartanalyzer

danrubel opened the pull request:
Now that `dartanalyzer` uses the default Flutter analysis options (`package:flutter/analysis_options_user.yaml`) to analyze a flutter project,
and a Dart SDK containing this functionality has rolled into Flutter...

Update `flutter analyze` to use `dartanalyzer` for everything except analyzing the flutter repository. 
* Contains many but not all of the changes in [the original PR](https://github.com/flutter/flutter/pull/7444)
* Fixes https://github.com/flutter/flutter/issues/8087 which appears to have regressed, and adds a test
* Incrementally [improves `flutter analyze`](https://github.com/flutter/flutter/issues/7274)
* One step closer to having `flutter analyze` always delegate to `dartanalyzer`
  so that flutter tools might someday not be tightly coupled to analyzer

@Hixie

Commit:
update flutter analyze to call dartanalyzer

Commit:
update flutter analyze to call dartanalyzer

Hixie commented on packages/flutter_tools/lib/src/commands/analyze_once.dart:
```diff
@@ -49,34 +50,48 @@ class AnalyzeOnce extends AnalyzeBase {
+    // Use dartanalyzer directly when possible
+    if (!flutterRepo) {
+      final String dartanalyzer = fs.path.join(
+          Cache.flutterRoot, 'bin', 'cache', 'dart-sdk', 'bin', 'dartanalyzer');
```
nit: newline before ) since there was a newline after the (

danrubel replied:
Fixed.

Hixie commented on packages/flutter_tools/lib/src/commands/analyze_once.dart:
```diff
@@ -49,34 +50,48 @@ class AnalyzeOnce extends AnalyzeBase {
+
+      if (cmd.length < 2 || currentPackage) {
+        final Directory projDir = await projectDirFor(fs.currentDirectory.absolute);
+        if (projDir != null) {
```
nit: no braces around one-line statements

danrubel replied:
Fixed.

Hixie commented on packages/flutter_tools/lib/src/commands/analyze_once.dart:
```diff
@@ -49,34 +50,48 @@ class AnalyzeOnce extends AnalyzeBase {
+      cmd.addAll(dartFiles.map((FileSystemEntity f) => f.path));
+
+      if (cmd.length < 2 || currentPackage) {
+        final Directory projDir = await projectDirFor(fs.currentDirectory.absolute);
```
nit: no abbreviations

danrubel replied:
Fixed.

Hixie commented on packages/flutter_tools/lib/src/commands/analyze_once.dart:
```diff
@@ -49,34 +50,48 @@ class AnalyzeOnce extends AnalyzeBase {
     final bool currentPackage = argResults['current-package'] && (argResults.wasParsed('current-package') || dartFiles.isEmpty);
```
I think we should remove `--current-directory` in this patch. This would simplify the logic here and is consistent with the long term plan in https://github.com/flutter/flutter/issues/7274.

danrubel replied:
Makes sense. Done.

Hixie commented on packages/flutter_tools/lib/src/commands/analyze_once.dart:
```diff
@@ -49,34 +50,48 @@ class AnalyzeOnce extends AnalyzeBase {
+        if (projDir != null) {
+          cmd.add((projDir).path);
+        }
+      } else if (currentDirectory) {
```
imho we should remove this here and just let people use `.` if they want this.

danrubel replied:
Removed.

Hixie commented on packages/flutter_tools/lib/src/commands/analyze_once.dart:
```diff
@@ -49,34 +50,48 @@ class AnalyzeOnce extends AnalyzeBase {
+        }
+      } else if (currentDirectory) {
+        cmd.add('.');
+      } else if (!cmd[1].startsWith(fs.currentDirectory.path)) {
```
I don't understand what this is doing. Why does the path of the first command line argument matter?

Hixie replied:
Also is `path` guaranteed to end in a separator?

danrubel replied:
1) If the files being analyzed are outside of the current directory hierarchy then dartanalyzer does not yet know how to find the `.packages` file. I'll add a comment to that effect.

2) Hmm... not sure. Modified the code to ensure that it does.

Hixie commented on packages/flutter_tools/lib/src/commands/analyze_once.dart:
```diff
@@ -49,34 +50,48 @@ class AnalyzeOnce extends AnalyzeBase {
+      stopwatch.stop();
+      final String elapsed = (stopwatch.elapsedMilliseconds / 1000.0).toStringAsFixed(1);
+      if (exitCode != 0)
+        throwToolExit('(Ran in ${elapsed}s)');
```
propagate the exit code?

danrubel replied:
Good idea. Done.

Hixie commented on packages/flutter_tools/lib/src/commands/analyze_once.dart:
```diff
@@ -49,34 +50,48 @@ class AnalyzeOnce extends AnalyzeBase {
-        if (isDartFile(entry)) {
-          dartFiles.add(entry);
-          foundOne = true;
+    // Use dartanalyzer directly when possible
```
this comment should elaborate on why it's not possible if flutterRepo is true.

danrubel replied:
Good point. Done.

Hixie commented on packages/flutter_tools/lib/src/commands/analyze_once.dart:
```diff
@@ -183,20 +196,32 @@ class AnalyzeOnce extends AnalyzeBase {
     }
   }
 
+  Future<Directory> projectDirFor(FileSystemEntity entity) async {
```
nit: no abbreviations (several cases in this method including the method name)

danrubel replied:
Renamed to `projectDirectoryContaining`.

Hixie commented on packages/flutter_tools/lib/src/commands/analyze_once.dart:
```diff
@@ -183,20 +196,32 @@ class AnalyzeOnce extends AnalyzeBase {
       else
         throwToolExit('(Ran in ${elapsed}s)');
     }
     if (argResults['congratulate']) {
```
Update the docs to mention that congratulate will be ignored if you're not using either `--watch` or `--flutter-repo`?

danrubel replied:
Help text updated.

danrubel commented on packages/flutter_tools/lib/src/commands/analyze_once.dart:
```diff
@@ -49,34 +50,48 @@ class AnalyzeOnce extends AnalyzeBase {
+      if (cmd.length < 2 || currentPackage) {
+        final Directory projDir = await projectDirFor(fs.currentDirectory.absolute);
+        if (projDir != null) {
+          cmd.add((projDir).path);
```
Artifact of refactoring tools. Removed.

danrubel commented on packages/flutter_tools/lib/src/commands/analyze_once.dart:
```diff
@@ -49,34 +50,48 @@ class AnalyzeOnce extends AnalyzeBase {
+    if (!flutterRepo) {
+      final String dartanalyzer = fs.path.join(
+          Cache.flutterRoot, 'bin', 'cache', 'dart-sdk', 'bin', 'dartanalyzer');
+      final List<String> cmd = <String>[dartanalyzer];
```
Renamed `cmd` to `arguments`. Hopefully that's more clear.

danrubel commented on packages/flutter_tools/lib/src/commands/analyze_once.dart:
```diff
@@ -49,34 +50,48 @@ class AnalyzeOnce extends AnalyzeBase {
+      final List<String> cmd = <String>[dartanalyzer];
+      cmd.addAll(dartFiles.map((FileSystemEntity f) => f.path));
+
+      if (cmd.length < 2 || currentPackage) {
```
I like that approach. Done.

Commit:
address comments

Commit:
address comments

Commit:
fix analyze_duplicate_names_test

Commit:
fix analyze_duplicate_names_test

Hixie commented on packages/flutter_tools/lib/src/commands/analyze_once.dart:
```diff
@@ -45,38 +46,62 @@ class AnalyzeOnce extends AnalyzeBase {
+    // Use dartanalyzer directly except when analyzing the Flutter repository.
+    // Analyzing the repository requires a more complex report than dartanalyzer
+    // currently supports (e.g. missing member dartdoc summary).
+    // TODO(danrubel) enhance dartanalyzer to provide this type of summary
```
nit: missing color after `TODO(danrubel)`

danrubel replied:
Fixed.

Hixie commented on packages/flutter_tools/lib/src/commands/analyze_once.dart:
```diff
@@ -45,38 +46,62 @@ class AnalyzeOnce extends AnalyzeBase {
+      final List<String> arguments = <String>[];
+      arguments.addAll(dartFiles.map((FileSystemEntity f) => f.path));
+
+      if (arguments.length < 1 || currentPackage) {
```
`arguments.length < 1` is better written `arguments.isEmpty`
Amusingly this is a creative-enough way to write `isEmpty` that the lint that looks for this kind of thing didn't catch it. :-)

danrubel replied:
Fixed.

Hixie commented on packages/flutter_tools/lib/src/commands/analyze_once.dart:
```diff
@@ -45,38 +46,62 @@ class AnalyzeOnce extends AnalyzeBase {
+          final Directory projDir = await projectDirectoryContaining(target);
+          if (projDir != null) {
+            final String packagesPath = fs.path.join(projDir.path, '.packages');
+            if (packagesPath != null) {
```
how can it be null?

danrubel replied:
Heh. Left over from refactoring. Removed `null` check.

Hixie commented on packages/flutter_tools/lib/src/commands/analyze_once.dart:
```diff
@@ -45,38 +46,62 @@ class AnalyzeOnce extends AnalyzeBase {
+          final String targetPath = arguments[0];
+          final FileSystemEntity target = await fs.isDirectory(targetPath)
+              ? fs.directory(targetPath) : fs.file(targetPath);
+          final Directory projDir = await projectDirectoryContaining(target);
```
nit: `projDir` is two abbreviations

danrubel replied:
Missed that one when renaming. Fixed.

Hixie commented on packages/flutter_tools/lib/src/commands/analyze_once.dart:
```diff
@@ -45,38 +46,62 @@ class AnalyzeOnce extends AnalyzeBase {
+        // In this situation, use the first file as a starting point
+        // to search for a "pubspec.yaml" and ".packages" file.
+        // TODO(danrubel): fix dartanalyzer to find the .packages file
+        if (!arguments[0].startsWith(currentDirectoryPath)) {
```
I think we should explicitly check that all the files would find the same `.packages` (or all would find no `.packages`?), and exit with a message saying that we don't yet support checking multiple packages at once if you're not also checking the flutter repo. I've definitely run into the case where I've tried to use `flutter analyze` to analyze two packages at once (an app I was working on and the support library I was writing for the app), and it would be massively confusing if that did something different depending on the order in which the paths were listed.

danrubel replied:
Good point. I simplified the code to always pass in the `.packages` file path and report an error as you suggested.

Hixie commented on packages/flutter_tools/test/analyze_once_test.dart:
```diff
@@ -0,0 +1,196 @@
+      toolExit: false,
+    );
+  });
+}
```
nice test!

Hixie commented on packages/flutter_tools/test/analyze_once_test.dart:
```diff
@@ -0,0 +1,196 @@
+        '[lint] Only throw instances of classes extending either Exception or Error',
+        '1 warning and 1 lint found.',
+      ],
+      // TODO(danrubel) fix dartanalyzer to have non-zero exit code
```
we should make sure to fix this asap because anyone depending on `flutter analyze` in their build bots will be skewered by this regression.

danrubel replied:
I updated this functionality to work around this issue so that `flutter analyze` will return a non-zero exit code even though `dartanalyzer` never does.

Hixie commented on packages/flutter_tools/lib/src/commands/analyze.dart:
```diff
@@ -29,7 +28,7 @@ class AnalyzeCommand extends FlutterCommand {
     // Not used by analyze --watch
-    argParser.addFlag('congratulate', help: 'Show output even when there are no errors, warnings, hints, or lints.', defaultsTo: true);
+    argParser.addFlag('congratulate', help: 'When analyzing the flutter repository, show output even when there are no errors, warnings, hints, or lints.', defaultsTo: true);
     argParser.addFlag('preamble', help: 'Display the number of files that will be analyzed.', defaultsTo: true);
```
looks like this also only works with --flutter-repo now

danrubel replied:
Good point. Help text updated.

Commit:
address comments and rework finding ".package" file

Commit:
address comments and rework finding ".package" file

Commit:
fix hint and disable locking for test

Commit:
fix hint and disable locking for test

Hixie commented on packages/flutter_tools/test/analyze_once_test.dart:
```diff
@@ -171,15 +172,11 @@ Future<Null> runCommand({
     await createTestCommandRunner(command).run(arguments);
     expect(toolExit, isFalse, reason: 'Expected ToolExit exception');
-  } on ToolExit catch (e) {
+  } on ToolExit catch (_) {
```
You can just remove the "catch" clause entirely (just `} on ToolExit {`)

danrubel replied:
Done.

Hixie commented on packages/flutter_tools/lib/src/commands/analyze.dart:
```diff
@@ -14,9 +14,11 @@ bool isDartFile(FileSystemEntity entry) => entry is File && entry.path.endsWith(
 class AnalyzeCommand extends FlutterCommand {
-  AnalyzeCommand({bool verboseHelp: false}) {
+  /// The working directory for testing analysis using dartanalyzer.
+  final Directory workingDirectory;
```
nit: constructor first

danrubel replied:
Fixed.

Hixie commented on packages/flutter_tools/lib/src/commands/analyze_once.dart:
```diff
@@ -24,7 +25,10 @@ typedef bool FileFilter(FileSystemEntity entity);
+  /// The working directory for testing analysis using dartanalyzer
+  final Directory workingDirectory;
+
+  AnalyzeOnce(ArgResults argResults, this.repoPackages, { this.workingDirectory }) : super(argResults);
```
nit: constructors first please

danrubel replied:
Sorry. Fixed.

Hixie commented:
LGTM modulo trivial style nits

Commit:
address comments

Commit:
address comments

Commit:
Merge branch 'master' into flutter-analyze-call-dartanalyzer

goderbauer committed:
add tasks for Windows bots in device lab (#8827)

pq committed:
Update to Dart SDK `1.23.0-dev.9.0`. (#8829)

Fixes: #8812

pq committed:
Escalate `missing_return` hints to warnings (#8825)

Fixes: https://github.com/flutter/flutter-intellij/issues/808

zanderso committed:
[fuchsia_reload] Find and attach to the first instance of the named app (#8830)

Hixie committed:
Increase timeout and verbosity to diagnose devicelab failure (#8832)

goderbauer committed:
Remove workaround for SIGINT (#8834)

It's no longer needed with the latest Dart SDK roll.

goderbauer committed:
fix build (#8842)

goderbauer committed:
Remove support for function keys. (#8840)

Function keys don't work great on any platform we support:
* Mac doesn't have first-class function keys.
* On Ubuntu: F1 opens the system help and F10 opens the file dialog.
* ... and Windows is a mess as well.

goderbauer committed:
Remove nasty ANSI workaround for Windows (#8839)

The problem has been fixed upstream in the Dart VM.

This simplifies our setup instructions on Windows (will update the wiki).
Furthermore, this also means that going forward there is no diffrence between PowerShell and Cmd for the Flutter experience on Windows.

cbracken committed:
Roll engine to 4a5a32466958dab49b9940e4528ee6d523f4a5ac (#8843)

abarth committed:
Add a couple dartdocs (#8841)

a14n committed:
replace const Duration() with Duration.ZERO (#8835)

a14n committed:
apply the upcoming rule unnecessary_this (#8838)

jakobr-google committed:
Make sure local.properties file is created before calling gradle. (#8847)

Fixes #8821

mravn-google committed:
Remove old platform messaging API (#8837)

Breaking change: removed deprecated methods of PlatformMessages, leaving only binary messaging there. All other use of platform communication now goes through PlatformMessageChannel and PlatformMethodChannels. Retained use of String and JSON codecs for now.

Companion engine PR: flutter/engine#3482

devoncarew committed:
update IntelliJ metadata (#8844)

abarth committed:
Ignore some missing plugins. (#8852)

Fixes #8850

pq committed:
Stop ignoring `strong_mode_down_cast_composite`s (#8783) (#8853)

Fixes: #8783

goderbauer committed:
make hot_mode_dev_cycle benchmark run in verbose mode (#8862)

pq committed:
Git ignore vscode `.history` files. (#8854)

The excellent `local-history` plugin for vscode, saves history in `.history` files; this ignores them.

See: https://marketplace.visualstudio.com/items?itemName=xyz.local-history

abarth committed:
Add dartdocs to scroll_view.dart (#8846)

abarth committed:
Add some `@required` annotations to transitions.dart (#8866)

Fixes #6382

Commit:
Verify that stateful hero state persists mid-flight (#8817)

abarth committed:
Rename ImageFit to BoxFit (#8871)

This machinery is useful for arbitrary boxes (e.g., with FittedBox).

Fixes #6463

Commit:
Now each Colors.foo constant is-a Color and a color swatch (#8833)

jason-simmons committed:
Handle malformed UTF-8 in logcat streams (#8864)

goderbauer committed:
Remove addFrameCallback (#8874)

Fixes #7793.

abarth committed:
Update engine (#8879)

abarth committed:
Adds support for custom fonts for icons (#8880)

This update adds support for custom fonts in icons. As an example, it
can be used with the icon-set from https://materialdesignicons.com

Thanks to @vlidholt for the original patch.

Fixes #4494
Fixes #3199

zanderso committed:
[fuchsia_reload] Add flag to override on-device binary name (#8873)

Commit:
Updating Cupertino Button MinSize to comply with iOS guidelines (#8881)

The recommended minimum tappable area is actually 44.0pt according to the [iOS Human Interface Guidline](https://developer.apple.com/ios/human-interface-guidelines/visual-design/layout/).

> Provide ample spacing for interactive elements. Try to maintain a minimum tappable area of 44pt x 44pt for all controls.

I've also added an override to allow for flexibility in situations where designers get naughty and have a desire to make everything super small.

abarth committed:
Switch from Size.isInfinite to Size.isFinite (#8868)

Fixes #4301

jakobr-google committed:
Remove legacy .apk build. (#8793)

* Remove legacy .apk build.

Print out an error message telling the user to upgrade the project if
it's not Gradle-based. Removed all the obvious traces of the legacy
build.

Added support for Dart VM kernel snapshots in Gradle builds.

Fixed Android installs to verify that the app is actually installed, and
not just rely on the presence of the .sha1 file.

jakobr-google committed:
Update complex_layout and microbenchmarks to Gradle build. (#8890)

goderbauer committed:
remove work around for https://github.com/dart-lang/sdk/issues/28599 (#8865)

abarth committed:
Add Image.memory for making images from typed data (#8877)

Fixes #7503
Fixes #8870

abarth committed:
Fixed extent lists cannot handle infinite extent (#8885)

Previously they tried to compute an integer target end index, but
integers can't represent infinity. Now we use null to represent
infinity.

Also, fix some similar issues with grids.

Fixes #8398

abarth committed:
Add WidgetTester.hasRunningAnimation (#8886)

Fixes #4017

szakarias committed:
Add integration test for the Platform Services app (#8826)

abarth committed:
Reference SizeChangedLayoutNotifier from SizeChangedLayoutNotification (#8887)

Fixes #7520

jason-simmons committed:
roll engine (#8917)

jason-simmons committed:
Render Material using a PhysicalModel with optional compositing (#8920)

Hixie committed:
Run both regular and --watch analysis on Travis (#8913)

We keep breaking --watch because we don't test it.

abarth committed:
Rename Dismissable to Dismissible (#8919)

The latter is the proper spelling.

Fixes #8883

abarth committed:
Document that Material clips (#8915)

Fixes #8519

abarth committed:
Add Design Discussion to State.build docs (#8916)

Fixes #8794

abarth committed:
Add a constant for padding in a Material ListView (#8918)

Fixes #8235

goderbauer committed:
Avoid downloading artifacts twice (#8872)

* Avoid downloading artifacts twice

* Don't update cache before `flutter upgrade` is executed (`flutter upgrade` might bump the engine version)
* Don't update cache twice during `flutter precache`

Fixes #8249.

* add test

abarth committed:
Align TextEditingValue and InputValue (#8922)

This patch prepares us to remove InputValue in favor of TextEditingValue.

asaarnak committed:
Update gradle wrapper and build-tools version for flutter create. (#8914)

* Update gradle wrapper and build-tools version.

Tested manually by "flutter create" with Idea "flutter run" and from android studio "run" android app.

* Update com.android.tools.build to 2.3.0

goderbauer committed:
Defer to operating system for whichAll (#8921)

The old `whichAll` implementation was not considering different extensions for executables on Windows. By defering to OS-built-in tools we avoid implementing it.

Fixes #8882.

Commit:
Make min/max fling velocity and min fling distance ScrollPhysics properties (#8928)

a14n committed:
add missing const keyword (#8931)

a14n committed:
use color.shadeXxx instead of color[Xxx] (#8932)

* use color.shadeXxx instead of color[Xxx]

* remove calls to .shade500 on MaterialColor

* remove calls to .shade200 on MaterialAccentColor

* fix test

mit-mit committed:
Revert "Update gradle wrapper and build-tools version for flutter create." (#8940)

* Revert "use color.shadeXxx instead of color[Xxx] (#8932)"

This reverts commit 578ca0a295c83d5be00de28f4ab62968a762df71.

* Revert "add missing const keyword (#8931)"

This reverts commit fac2fac1d67708da14364dc90f40e06130da9d21.

* Revert "Make min/max fling velocity and min fling distance ScrollPhysics properties (#8928)"

This reverts commit dac80aac89959a57194d96ffe06211b3faffc907.

* Revert "Defer to operating system for whichAll (#8921)"

This reverts commit 14933de986b1264015ea69442603520cff969650.

* Revert "Update gradle wrapper and build-tools version for flutter create. (#8914)"

This reverts commit deb71cc6dcf45297ab4fb6e6ccf3d14fb05be9ab.

szakarias committed:
Open storyboard documents in Xcode 7 format (#8946)

szakarias committed:
Refactor platform_services sample (#8949)

* Refactor platform_services sample

* removed exception

jakobr-google committed:
Remove unnecessary Android SDK checks. (#8926)

* Remove unnecessary Android SDK checks.

* Remove unused accessors, skip replay test.

jakobr-google committed:
Add more logging to diagnose test failures. (#8951)

Commit:
Fix selected tab indicator tap animation (#8939)

Hixie committed:
Make the `flutter test` tests resilient to startup lock (#8941)

goderbauer committed:
Fix various gradle paths for Windows (#8935)

goderbauer committed:
Print error message if a package is skipped during DevFS sync. (#8960)

* Print error message if a package is skipped during DevFS sync.

* review comments

* more review comments

* fix test

goderbauer committed:
Report detailed OS version string to analytics (#8934)

collinjackson committed:
Improvements to CircleAvatar text color (for #8895 codelab) (#8936)

* Improvements to CircleAvatar default color (for codelab)

* use a transparent white instead of a solid one

* Fix analyzer errors

* fix tests

kevmoo committed:
Teach flutter_tools to populate PUB_ENVIRONMENT (#8962)

* Teach flutter_tools to populate PUB_ENVIRONMENT

Will allow telemetry reporting on pub.dartlang.org once
flutter moves to 1.23.0-dev.10.0

* review changes

cbracken committed:
Build Flutter app as a framework on iOS (#8971)

**THIS IS A BREAKING CHANGE.** See below for migration steps for
existing projects.

Previously, Flutter app code was built as a raw dylib on iOS.  Dynamic
libraries outside of a framework bundle are not supported on iOS, except
for the system Swift libraries provided by Xcode.

See:
https://developer.apple.com/library/content/technotes/tn2435/_index.html#//apple_ref/doc/uid/DTS40017543-CH1-TROUBLESHOOTING_BUNDLE_ERRORS-EMBEDDED__DYLIB_FILES

* Migrates Xcode build from app.dylib to App.framework
* Migrates flutter create template
* Migrates example projects

Migration steps for existing projects
=====================================

The following steps should be taken from the root of your Flutter
project:

1. Edit `ios/.gitignore`: add `/Flutter/App.framework` on a new line.
2. In the Xcode project navigator, remove `app.dylib` from the Flutter
   folder. Delete this file from the `ios/Flutter` directory in your project.
3. Run a build to generate `ios/Flutter/App.framework`. From the command
   line, run `flutter build ios`. If you have not configured app signing
   in Xcode, an alternative method is to open the simulator, then run
   `flutter run -d iP`.
4. In the Xcode project navigator, select the `Runner` project. In the
   project settings that are displayed in the main view, ensure that the
   `Runner` target is selected. You can verify this by exposing the
   sidebar using the [| ] icon in the upper-left corner of the main
   view.
5. Select the *General* tab in the project settings. Under the
   *Embedded Binaries* section, click '+' to add `App.framework`. In the
   sheet that drops down, click the *Add Other...* button. Navigate to
   the `ios/Flutter` directory and select `App.framework`. Click *Open*.
   In the sheet that drops down, select *Create folder references*, then
   click *Finish*.
6. In the project settings, verify that `App.framework` has been added to the
   *Embedded Binaries* and *Linked Frameworks and Libraries* lists.
7. In the Xcode project navigator, drag `App.framework` under the
   Flutter folder.
8. In the Xcode project navigator, select `Flutter` then from the
   *File* menu, select *Add Files to "Runner"...*. Navigate to the
   `ios/Flutter` directory, select `AppFrameworkInfo.plist` and click
   the *Add* button.
9. From the command line, in your project directory, run
   `flutter build clean`, then `flutter run`.

At this point your project should be fully migrated.

mravn-google committed:
Engine roll to get #3498 (#8966)

mravn-google committed:
Updated complex_layout ios project according to #8971 (#8974)

mravn-google committed:
Updated microbenchmark ios project according to #8971 (#8976)

szakarias committed:
Add start-up test for flutter_view sample (#8892)

* Add start-up test for flutter_view sample

* add reportMetrics flag to startup test

* rewording in manifest.yaml

mit-mit committed:
Update links to platform services -> channels (#8975)

jakobr-google committed:
Automatically wire dependencies for native plugins (#8891)

Go through all packages brought in by pub, and write the name and path of every one that is a flutter plugin into .flutter-plugins.

In android/settings.gradle and ios/Podfile, read in .flutter-plugins, if that file exists. The Android / iOS code from the plugins is automatically added as dependencies of the native code of the app.

szakarias committed:
Remove flutter_view_ios__start_up.dart (#8978)

pylaligand committed:
Generate a depfile for assets of a flutter_app. (#8961)

goderbauer committed:
Use Java bundled with Android Studio for gradle (#8965)

* Use Java bundled with Android Studio for gradle

* review comments

xster committed:
Remove various roboto font logic (#8350)

cbracken committed:
Use const for string literals where possible (#8980)

a14n committed:
use_rethrow_when_possible (#8985)

lukef committed:
Adds highlight + splash color properties to material button (#8643)

* adds highlight + splash color properties to material button

* add states to raised button + flat button

* fixes an issue where the custom highlight would not be properly set

* Add preliminary tests for theme highlight + splash and via theme

* documentation changes. tweaks per code review feedback

* comment out unused variable

* remove unused import

* documentation updates. removed commented code

yjbanov committed:
Xcode 8-compatible provisioning profiles in devicelab tests (#8987)

szakarias committed:
Revert "Remove flutter_view_ios__start_up.dart (#8978)" (#8996)

This reverts commit 38e4e48d06cd23cecc3168e9c89d686e88a683ef.

szakarias committed:
Remove hello_services sample (#8995)

szakarias committed:
Update the examples README file (#8994)

* Update the examples README file

* Addressed comment

szakarias committed:
Patch Debug/Release.xcconfig files in devicelab tests (#9000)

Commit:
Tweaked the appaearance of the cards demo (#8964)

yjbanov committed:
detailed error message re missing xcconfigs (#9001)

johnmccutchan committed:
Attempt to fix devfs test (#9007)

Commit:
Eliminated DrawerItem, use ListTile instead (#8992)

collinjackson committed:
Fix #9003 Make DrawerHeader margin configurable (#9006)

goderbauer committed:
Fix crash when old Android Studio (<2.2.0) is installed (#9009)

Old Android Stuio versions pior to 2.2.0 didn't come with Java bundled. Make sure flutter doesn't crash when we try to determine the Java version of those old Android Studio installations.

lyrachord committed:
typo spontaneously (#9013)

abarth committed:
Add fontFamily to ThemeData constructor (#9016)

Otherwise it's somewhat verbose to configure all the text themes.

abarth committed:
Fix the parameter names for computeMaxIntrinsicHeight and computeMinIntrinsicHeight (#9017)

These parameters are the width, not the height.

jakobr-google committed:
Make pod install a long-running step. (#8999)

The very first time `pod install` is invoked, CocoaPods downloads the master spec repository, which takes quite a while. Before this change, the build appeared to have stalled. With this change, at least the spinner is moving.

Added `pod setup` to the install instructions for CocoaPods, so the spec repo is downloaded while setting up Flutter, instead of during the first build.

szakarias committed:
Rename platform_services to platform_channel (#9018)

* Rename platform_services to platform_channel

* rename in README

* renamed tests

* update modules.xml

jakobr-google committed:
iOS: Make flutter doctor unhappy if CocoaPods is missing. (#8979)

szakarias committed:
remove platform_channel/android/buildSrc (#9020)

szakarias committed:
update platform_channel package name (#9021)

* update platform_channel package name

* lowercase name

vlidholt committed:
Fixes issue with removing and re-adding a toolbar which is connected to the same TabBarController (#9015)

tvolkert committed:
Update devfs_test to use MemoryFileSystem (#8811)

goderbauer committed:
Disable flaky WIndows test :( (#9029)

lukef committed:
Add border radius to Ink widgets (#8693)

* adds border radius to ink widgets. sets default ink border radius for material buttons with no background colors

* tidying up code

* add ink test stub

* remove unused import

Commit:
Shrine gallery demo: support for landscape layout (#9025)

collinjackson committed:
Include fragment support in Android build to allow use of FragmentActivity in FlutterActivity (#9036)

Commit:
Added a TabBar regression test (#9038)

collinjackson committed:
Revert "Include fragment support in Android build to allow use of FragmentActivity in FlutterActivity (#9036)" (#9041)

This reverts commit 6bf0ceb02bda8b9a93569b32a7c726cc560b2176.

collinjackson committed:
Revert "Revert "Include fragment support in Android build to allow use of FragmentActivity in FlutterActivity (#9036)"" (#9046)

Increased the time limit slightly to allow the microbenchmark test time to finish.

collinjackson committed:
Revert "Revert "Revert "Include fragment support in Android build to allow use of FragmentActivity in FlutterActivity (#9036)"" (#9046)" (#9047)

Test is still failing with increased timeout.

This reverts commit 864b3c37c65a4294735bbdb041d17b112a2659cc.

abarth committed:
Update and modernize the flutter_markdown package (#9044)

We now use modern scrolling machinery and patterns. The API should also be
easier to maintain over time.

Fixes #6166
Fixes #2591
Fixes #3123

Commit:
Merge branch 'flutter-analyze-call-dartanalyzer' of github.com:danrubel/flutter into flutter-analyze-call-dartanalyzer

danrubel commented:
So.... this is what happens when I rebase and push.
Note to self... don't use rebase in this situation. :-)

danrubel commented:
Closing this in favor of the identical PR... https://github.com/flutter/flutter/pull/9049
